test/Semantic/Spec.hs

```diff
@@ -24,5 +25,22 @@ spec = parallel $ do
     it "renders with the specified renderer" $ do
       output <- fmap runBuilder . runTaskOrDie $ parseTermBuilder TermSExpression [methodsBlob]
       output `shouldBe` "(Statements\n  (Method\n    (Empty)\n    (Identifier)\n    (Statements)))\n"
+
+  describe "gitParsing" $ do
+    it "parses a git output string" $ do
+      let input = "100644 tree ThisIsTheOid\t/this/is/the/path"
+      let expected = TreeEntry NormalMode TreeObject (OID "ThisIsTheOid") "/this/is/the/path"
+      parseEntry input `shouldBe` expected
+
+    it "parses nonsense into a default value" $ do
```

Comment:
I don’t think we need this one, though: if we get nonsense I want the parser to complain loudly, and we can handle failure at a higher-level.